### PR TITLE
Add support for new genedit functions

### PR DESCRIPTION
**Pull request by jlorentson**

### CODE REVIEW

File: src/test/java/com/imsweb/validation/functions/MetafileContextFunctionsTest.java [hunk 1]
```diff
@@ -1166,6 +1166,217 @@ public void testGEN_ILOOKUP() throws ConstructionException {
...
+        table.add(row2);
+        List<Object> row3 = new ArrayList<>();
+        row3.add("18");
+        row3.add("Another Bones");
+        row3.add("C400");
+        row3.add("C409");
+        table.add(row3);
+
+        List<Object> columns = new ArrayList<>();
+        Assert.assertTrue(_functions.GEN_SQLLOOKUP(table, columns, ""));
```
Comment (jlorentson) [inline src/test/java/com/imsweb/validation/functions/MetafileContextFunctionsTest.java]: It's the empty column list that always returns `true`; further down after `columns` is populated, this assert passes: `Assert.assertFalse(_functions.GEN_SQLLOOKUP(table, columns, ""));`

When an empty string representing column names to search is passed to `SQLLOOKUP `or `SQLRANGELOOKUP`, the Genedits throws an error in a popup: 
image
But if you click "OK," the Notes/Results pane says: "The edit PASSED"
This is very strange, but short of throwing an exception, I'm not sure of the best way to handle/emulate it.

File: src/main/java/com/imsweb/validation/functions/MetafileContextFunctions.java [hunk 2]
```diff
@@ -975,6 +975,88 @@ else if (index instanceof Set)
...
+                return false;
+        }
+
+        int indexOfLargest = -1;
+        String largest = "";
+        for (int r = 1; r < table.size(); r++) {
+            StringBuilder rowVal = new StringBuilder("");
+            for (int c : colNumbers)
+                rowVal.append(GEN_TO_STRING(table.get(r).get(c)));
+            if (StringUtils.isNumeric(val) && StringUtils.isNumeric(rowVal)) {
```
Comment (jlorentson) [inline src/main/java/com/imsweb/validation/functions/MetafileContextFunctions.java]: String.compare("2", "10") returns 2 > 10. Their code returns 2 < 10. In the case where the values are alphanumeric, it uses String compare (or something similar), so "A2" > "A10" for them, but for straight numbers we need separate logic.

File: src/main/java/com/imsweb/validation/functions/MetafileContextFunctions.java [hunk 3]
```diff
@@ -975,6 +975,88 @@ else if (index instanceof Set)
...
+        int[] colNumbers = new int[columnNames.size()];
+        for (int i = 0; i < colNumbers.length; i++) {
+            colNumbers[i] = table.get(0).indexOf(columnNames.get(i));
+            if (colNumbers[i] == -1)
+                return true;
+        }
+
+        for (int r = 1; r < table.size(); r++) {
+            int index = 0;
+            boolean match = false;
```
Comment (jlorentson) [inline src/main/java/com/imsweb/validation/functions/MetafileContextFunctions.java]: Yep. I changed the guts of that `for`-loop to match the other method, but this one doesn't need those variables.

File: src/test/java/com/imsweb/validation/functions/MetafileContextFunctionsTest.java (hunk 1 above)
Comment (depryf) [inline src/test/java/com/imsweb/validation/functions/MetafileContextFunctionsTest.java]: We are good. I thought the empty value caused it to return true, didn't realize the columns were empty as well. This is resolved.

File: src/main/java/com/imsweb/validation/functions/MetafileContextFunctions.java (hunk 2 above)
Comment (depryf) [inline src/main/java/com/imsweb/validation/functions/MetafileContextFunctions.java]: Sounds good. As long as we agree with their behavior :-)

**Pull request merged**